code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -3980,21 +3980,40 @@
 	description = "A powerful sedative."
 	reagent_state = REAGENT_STATE_SOLID
 	color = "#000067" //rgb: 0, 0, 103
-	data = 1 //Used as a tally
+	// There used to be a bug: if someone was injected with chloral once,
+	// and then injected with chloral a second time, this person would
+	// wake up for the duration of a Mob Subsystem tick.
+	// Now, how could this be? This proc was very clearly only ever increasing
+	// the sleeping var, never decreasing it. As it turns out,
+	// /datum/proc/reagents/proc/trans_to takes an argument called
+	// "preserve_data", which, if not null, tells
+	// /datum/reagents/proc/add_reagent to set the data var of
+	// the destination reagent to the one of the source reagent.
+	// add_reagent, in turn, would only override the old data with the new if
+	// the new data was not null. Since /datum/reagent/chloralhydrate/data
+	// was set to 1, it did end up resetting the data var of the existing
+	// chloralhydrate in the spessman's body, waking them up until the
+	// following tick. So that's the reason this var is null here.
```

Comment:
the length of this comment is deity tier
it's not a bad idea to preserve some info for future coders but I think you can get this down to 4 lines